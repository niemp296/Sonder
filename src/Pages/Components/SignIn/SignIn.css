.outerContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 5%;
    background-color: white;
}

.registrationForm{
    min-width: 450px;
    max-width: 60%;
    height: 60%;
    display: flex;
    flex-direction: column;
    padding: 20px 40px;
    border-radius: 30px;
    box-shadow: 0px 10px 50px rgb(207, 207, 207);
    background-color: white;
    padding-bottom: 2%;
}

.title {
    font-family: 'Baskerville';
    font-size: 4em;
    color: #E58C8A;
    margin-bottom: 7%;
    margin-top: 5%;
    font-weight: lighter;
}

form {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
}

h1 {
    text-align: center;
    width: 100%;
    color: #343434;
    font-weight: bold;
    font-family: 'Poppins';
    font-size: 1em;
}

label {
    font-size: 1em;
    margin-bottom: 0.25em;
    color: #343434;
    font-weight: bold;
    font-size: 14px;
    line-height: 21px;
}

input {
    padding: 10px 10px;
    border-radius: 5px;
    outline: none;
    border: 1px solid #cfcfcf;
}

input::placeholader {
    font-size: 1.2em;
    font-weight: lighter;
    color: #343434;
}

input.error {
    border: 1px solid red;
}

.errorMessage {
    color: red;
    font-size: 0.75em;
    display: relative;
}

.email,
.password {
    display: flex;
    flex-direction: column;
    margin-bottom: 15px;
}

.email,
.password {
    width: 100%;
}

.signIn {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.signIn button {
    background-color: #E58C8A;
    color: #fff;
    border: 2px solid #fff;
    width: 100%;
    margin-top: 1em;
    padding: 8px 0px;
    font-size: 1.5em;
    font-weight: bold;
    letter-spacing: 1px;
    margin-bottom: 0.7em;
}

.signIn small {
    color: #343434;
    font-weight: '600';
    font-size: 18px;
    text-transform: capitalize;
    font-family: 'Poppins';
    font-style: normal;
}

