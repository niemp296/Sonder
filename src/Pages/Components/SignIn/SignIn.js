import React, { Component } from "react";
import "./SignIn.css";
import Link from '@material-ui/core/Link';
import Header from '../Header/Header';
import axios from 'axios';
import { Redirect } from 'react-router-dom'

const checkEmailFormat = RegExp(
  /^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
);

const formValid = ({ formErrors, ...rest }) => {
  let valid = true;

  // validate form errors being empty
  Object.values(formErrors).forEach(val => {
    val.length > 0 && (valid = false);
  });

  // validate the form was filled out
  Object.values(rest).forEach(val => {
    val === null && (valid = false);
  });

  return valid;
};

class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: null,
      password: null,
      formErrors: {
        email: "",
        password: ""
      },
      isSignedIn: false,
      user_id: 0
    };
  }

  handleSubmit = e => {
    e.preventDefault();

    if (formValid(this.state)) {
      if (this.state.firstName !== "") {
        axios.post('http://localhost:5000/api/signIn', this.state)
            .then(response => {
                console.log(response);

              if(response.data === "404" || response.data === 404){
                alert("User does not exists")
              }
              else if(response.data === "400"){
                alert("Wrong password")
              }
              else {
                this.setState({user_id: response.data});
                this.setState({isSignedIn: true});
                console.log("sign_in is succesful");
              }
        })
        .catch(function(error){
          //Perform action based on error
            console.log(error);
        });
      }
    } else {
      console.error("FORM INVALID - DISPLAY ERROR MESSAGE");
    }
    console.log("end of sign in method")
  };

  handleChange = e => {
    e.preventDefault();
    const { name, value } = e.target;
    let formErrors = { ...this.state.formErrors };

    switch (name) {
      case "email":
        formErrors.email = checkEmailFormat.test(value) ? "" : "invalid email address";
        break;
      case "password":
        formErrors.password =
          value.length < 6 ? "minimum 6 characaters required" : "";
        break;
      default:
        break;
    }

    this.setState({ formErrors, [name]: value });
  };

  render() {
    const { formErrors } = this.state;

    if(this.state.isSignedIn){
      var path_name = '/Account:' + this.state.user_id;
      return <Redirect to = {{pathname: path_name}} />;
    }
    return (
        <div>
            <Header />     
        <div className="outerContainer">
            <div className="registrationForm">
                <h1 className="title">
                    Welcome back!
                </h1>
            <form action ="http://localHost:5000/api/signIn" method ="post" onSubmit={this.handleSubmit} noValidate>
                <div className="email">
                <label htmlFor="email">Email</label>
                <input
                    className={formErrors.email.length > 0 ? "error" : null}
                    placeholder="Email"
                    type="email"
                    name="email"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.email.length > 0 && (
                    <span className="errorMessage">{formErrors.email}</span>
                )}
                </div>
                <div className="password">
                <label htmlFor="password">Password</label>
                <input
                    className={formErrors.password.length > 0 ? "error" : null}
                    placeholder="Password"
                    type="password"
                    name="password"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.password.length > 0 && (
                    <span className="errorMessage">{formErrors.password}</span>
                )}
                </div>
                <div className="signIn">
                <button type="submit">Sign In</button>
                    <Link href="/sign-up" variant="body">
                    <small>  {"Don't have your account? Sign Up"} </small>
                    </Link>
                    <Link href="/forgot-password" variant="body">
                        <small>  {"Forgot your password?"} </small>
                    </Link>
                </div>
            </form>
            </div>
        </div>
      </div>
    );
  }
}

export default SignIn;