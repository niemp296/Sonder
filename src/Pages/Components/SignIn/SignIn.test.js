import React from 'react';
import { mount, shallow , configure} from 'enzyme';
import SignIn from './SignIn';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe("Test case for SignIn layout", () => {
    it("should render sign in page", () => {
        const wrapper = shallow(<SignIn />);
    });

    it("should render initial layout", () => {
        // when
        const component = shallow(<SignIn/>);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });
});


describe('Test case for testing SignIn',() =>{
    let wrapper;
    const preventDefault = jest.fn();
    
    it('email check',()=>
    {
        wrapper = mount(<SignIn/>);
        wrapper.find('input[type="email"]').simulate('change', {target: {name: 'email', value: 'irennanicole1@gmail.com'}});
        expect(wrapper.state('email')).toEqual('irennanicole1@gmail.com');
    })
    it('password check',()=>{
        wrapper = mount(<SignIn/>);
        wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: '111111'}});
        expect(wrapper.state('password')).toEqual('111111');
    })
    it('login check with data',()=>{
        wrapper = mount(<SignIn/>);
        wrapper.find('input[type="email"]').simulate('change', {target: {name: 'email', value: 'irennanicole1@gmail.com'}});
        wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'phuongn123'}});
        wrapper.find('button').simulate('click', { preventDefault });
        wrapper.find('form').simulate('submit', { preventDefault });
        expect(wrapper.state('isSignedIn')).toBe(false);
    })
})


