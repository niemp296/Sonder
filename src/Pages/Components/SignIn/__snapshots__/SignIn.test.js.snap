// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for SignIn layout should render initial layout 1`] = `
Array [
  <div>
    <Header />
    <div
      className="outerContainer"
    >
      <div
        className="registrationForm"
      >
        <h1
          className="title"
        >
          Welcome back!
        </h1>
        <form
          action="http://localHost:5000/api/signIn"
          method="post"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="email"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className={null}
              name="email"
              noValidate={true}
              onChange={[Function]}
              placeholder="Email"
              type="email"
            />
          </div>
          <div
            className="password"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className={null}
              name="password"
              noValidate={true}
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="signIn"
          >
            <button
              type="submit"
            >
              Sign In
            </button>
            <ForwardRef(WithStyles)
              href="/sign-up"
              variant="body"
            >
              <small>
                  
                Don't have your account? Sign Up
                 
              </small>
            </ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              href="/forgot-password"
              variant="body"
            >
              <small>
                  
                Forgot your password?
                 
              </small>
            </ForwardRef(WithStyles)>
          </div>
        </form>
      </div>
    </div>
  </div>,
]
`;
